internals/src/Tizen.Applications.ThemeManager/Tizen.Applications.ThemeManager/ThemeLoader.cs

```diff
@@ -0,0 +1,217 @@
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        public Theme LoadTheme(string id)
+        {
+            IntPtr _themeHandle;
+
+            Interop.ThemeManager.ErrorCode err = Interop.ThemeManager.LoaderLoadTheme(_loaderHandle, id, out _themeHandle);
+            if (err != Interop.ThemeManager.ErrorCode.None)
+            {
+                Interop.ThemeManager.ThemeManagerErrorFactory.GetException(err, "Failed to load theme");
+            }
+
+            return new Theme(_themeHandle);
+        }
+
+        /// <summary>
+        /// Gets bundle of theme IDs.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <exception cref="ArgumentException">Thrown when failed because of an invalid argument.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when failed because of out of memory.</exception>
+        public IEnumerable<string> QueryIds()
+        {
+            IntPtr ids;
+            string[] stringArray;
+            int count;
+
+            Interop.ThemeManager.ErrorCode err = Interop.ThemeManager.LoaderQueryId(_loaderHandle, out ids, out count);
+            if (err != Interop.ThemeManager.ErrorCode.None)
+            {
+                Interop.ThemeManager.ThemeManagerErrorFactory.GetException(err, "Failed to query ids");
+            }
+
+            IntPtrToStringArray(ids, count, out stringArray);
+            return stringArray;
+        }
+
+        static void IntPtrToStringArray(IntPtr unmanagedArray, int size, out string[] managedArray)
+        {
+            managedArray = new string[size];
+            IntPtr[] IntPtrArray = new IntPtr[size];
+            Marshal.Copy(unmanagedArray, IntPtrArray, 0, size);
+            for (int iterator = 0; iterator < size; iterator++)
+            {
+                managedArray[iterator] = Marshal.PtrToStringAnsi(IntPtrArray[iterator]);
+            }
+        }
+
+        /// <summary>
+        /// Releases all resources used by the ThemeLoader class.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases the unmanaged resources used by the ThemeLoader class specifying whether to perform a normal dispose operation.
+        /// </summary>
+        /// <param name="disposing">true for a normal dispose operation; false to finalize the handle.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (_loaderHandle != IntPtr.Zero)
+            {
+                Interop.ThemeManager.LoaderDestroy(_loaderHandle);
+                _loaderHandle = IntPtr.Zero;
+            }
+
+            if (_currentTheme != null)
```

Comment:
You should released a managed object only when disposing was true
```suggestion
            if (disposing && _currentTheme != null)
```